47.md

```diff
@@ -135,6 +137,95 @@ The **client** should then store this connection and use it when the user wants
 nostr+walletconnect://b889ff5b1513b641e2a139f661a661364979c5beee91842f8f0ef42ab558e9d4?relay=wss%3A%2F%2Frelay.damus.io&secret=71a8c14c1407c113601079c4302dab36460f0ccd0ad506f1f2dc73b5100e4f3c
 ```
 
+## Client-Created Secret
+
+The client MAY generate its own secret and co-ordinate only public keys with the wallet service so that secrets are never exposed.
+
+### HTTP Confirmation Page
+
+A **wallet service** MAY host a publicly accessible confirmation page that the **client** may open (e.g. in a webview or popup window) and allow the **user** to confirm the creation of a new connection, with the following parameters:
+
+- `pubkey` Required. The corresponding pubkey of the secret generated by the **client**.
+
+The **wallet service** MAY ignore all the below optional parameters.
+
+- `name`: Optional. The name of the client app.
+- `icon`: Optional. The URL of an icon of the client app to display on the confirmation page.
+- `return_to` Optional. URI to open after the connection is created. The **wallet service** SHOULD add the `relay_url`, `wallet_pubkey` and `lud16` parameters to the URI.
+- `expires_at` Optional. The connection cannot be used after this date. Unix timestamp in seconds.
+- `max_amount` Optional. The maximum amount in millisats that can be sent per renewal period
+- `budget_renewal` Optional. The reset the budget at the end of the given budget renewal. Can be `never` (default), `daily`, `weekly`, `monthly`, `yearly`
+- `request_methods` Optional. The url encoded, space separated list of request types that you need permission for. For example: `..&request_methods=pay_invoice%20get_balance`
+- `notification_types` Optional. The url encoded, space separated list of notification types that you need permission for. For example: `..&notification_types=payment_received%20payment_sent`
+- `isolated` Optional. The makes an isolated app connection / sub-wallet with its own balance and only access to its own transaction list. e.g. `&isolated=true`.
+- `metadata` Optional. Url encoded, JSON-serialized metadata that describes the app connection.
+
+The **user** MUST be presented with a confirmation page to be able to review and approve or decline the connection.
```

Comment:
Any thoughts on *how* this happens? For UMA auth, we use the user's UMA (or lightning address) fetch a configuration document similar to the OIDC discovery doc. From there, we just read the `authorization_endpoint` field to decide what url to open. It's really similar to how oauth works in this regard and allows the user to just type a lightning address to open the right endpoint (or detect that this method won't work).

See https://docs.uma.me/uma-auth/uma-auth-client/client-manual-implementation#oauth-20-exchange

Reply:
Our current solution is to have a list of wallets to pick from (e.g. Bitcoin Connect has a list of wallets it supports and knows what flows to use, and in the http case what url to use).

I think there are pros and cons of both approaches though.